R/check.R

```diff
@@ -10,115 +10,48 @@
 #'
 #' @family check
 check_quantiles <- function(posterior, req_probs = c(0.5, 0.95, 0.2, 0.8)) {
-  if (any(req_probs <= 0) || any(req_probs >= 1)) {
-    stop("Please provide probabilities as numbers between 0 and 1.")
-  }
-  cols <- colnames(posterior)
-  if (sum(cols %in% paste0("q", req_probs * 100)) != length(req_probs)) {
-    stop(
-      "Following quantiles must be present (set with probs): ",
-      toString(req_probs)
-    )
-  }
-  return(invisible(NULL))
+  stopifnot(
+    "Please provide probabilities as numbers between 0 and 1." =
+    all(data.table::between(req_probs, 0, 1, incbounds = FALSE))
+  )
+  return(coerce_dt(
+    posterior, required_cols = sprintf("q%g", req_probs * 100),
+    copy = FALSE
+  ))
 }
 
-#' Check Report and Reference Dates are present
+#' Check for and Format Report and Reference Dates
 #'
-#' @param obs An observation `data.frame` containing \code{report_date} and
-#' \code{reference_date} columns.
+#' @param obs An object that will be [coerce_dt()]d in place; must have
+#' `report_date` and `reference_date` columns.
 #'
-#' @return Returns the input `data.frame` with dates converted to date format
-#' if not already.
+#' @return The `obs` object, modified in place (if necessary), where
```

Comment:
it looks like it never copies so this seems inaccurate.

Reply:
well, if report_date and reference_date are already IDates, this is a no-op

Reply:
but i think i'm going to move date in coerce_dt